src/coreclr/inc/profilepriv.inl

```diff
@@ -1762,6 +1762,16 @@ inline void ProfControlBlock::EventPipeProviderCreated(EventPipeProvider *provid
                        provider);
 }
 
+// Contrary to other profiler callbacks, EnumerateGCHeabObjectsCallback is a wrapper enabling object inspection
+// through the direct (non ICorProfilerCallback) callback in ProfToEEInterfaceImpl::EnumerateGCHeapObjects.
+// As such, this callback is not concerned with calling all attached profilers through DoProfilerCallback,
+// and instead, invokes the callback directly on the EEToProfInterfaceImpl instance.
+inline HRESULT ProfControlBlock::EnumerateGCHeapObjectsCallback(EEToProfInterfaceImpl *profInterface, ObjectCallback callback, void* callbackState)
+{
+    return profInterface->EnumerateGCHeapObjectsCallback(callback, callbackState);
```

Comment:
Would it be better to make this call directly from `ProfToEEInterfaceImpl::EnumerateGCHeapObjects` and delete this method?

Reply:
Ah right, since we can grab the EEToprofInterfaceImpl directly, we don't need to route through `g_profControlBlock`. Thanks for the catch!

Reply:
Removed the ProfControlBlock changes